File: Java/프로그래머스/1/42862. 체육복/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] student = new int[n];//체육복 있음 0
        
        for(int l : lost) {//체육복 없음 -1
            student[l-1]--;
        }
        for(int r : reserve) {//체육복 여분 있음 1
            student[r-1]++;
        }
        for(int i = 0; i < n; i++) {
            if(student[i] < 0) {
                if(i>0 && student[i-1] > 0) {
                    student[i]++;
                    student[i-1]--;
                } else if(i< n-1 && student[i+1] > 0){
                    student[i]++;
                    student[i+1]--;
                }
            }
            if(student[i] > -1) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: Kotlin/프로그래머스/1/42862. 체육복/체육복.kt
class Solution {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        var students = Array(n){0}
        lost.forEach{students[it-1]--}
        reserve.forEach{students[it-1]++}
        
        students.forEachIndexed{
            i, v -> 
            if(v==-1) {
                if(i>0 && students[i-1] > 0) {
                    students[i]++
                    students[i-1]--
                } else if(i<students.lastIndex && students[i+1] > 0) {
                    students[i]++
                    students[i+1]--
                }
            }
        }
        
        return students.count{it>-1}
    }
}